code/modules/power/tesla/coil.dm

```diff
@@ -30,15 +30,20 @@
 
 	if(W.iswrench())
 		playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
-		user << "<span class='notice'>You [anchored ? "unfasten" : "fasten"] [src] to the flooring.</span>"
+		to_chat(user, "<span class='notice'>You [anchored ? "unfasten" : "fasten"] [src] to the flooring.</span>")
 		anchored = !anchored
 		if(!anchored)
 			disconnect_from_network()
 		else
 			connect_to_network()
 		return
 
-/obj/machinery/power/tesla_coil/tesla_act(var/power)
+/obj/machinery/power/tesla_coil/tesla_act(var/power, var/melt = FALSE)
+	if(melt)
```

Comment:
Duplication of code detected. Is it not possible to add this clause to the original definition of `tesla_act`, and make it apply to all objects? The override definitions can then choose how they want to pass the melt argument down.